rest-api/api/dv_order_api.py

```diff
@@ -96,13 +113,19 @@ def put(self, bo_id):  # pylint: disable=unused-argument
     return method(resource, bo_id)
 
   def _put_supply_request(self, resource, bo_id):
-    fhir_resource = SimpleFhirR4Reader(resource)
-    barcode_url = None
-    if fhir_resource.extension.get(url=VIBRENT_FULFILLMENT_URL).valueString == 'shipped':
-      barcode_url = fhir_resource.extension.get(url=VIBRENT_BARCODE_URL).url
-    pid = fhir_resource.contained.get(
-      resourceType='Patient').identifier.get(system=VIBRENT_FHIR_URL + 'participantId')
-    p_id = from_client_participant_id(pid.value)
+
+    # handle invalid FHIR documents
+    try:
+      fhir_resource = SimpleFhirR4Reader(resource)
+      barcode_url = None
+      if fhir_resource.extension.get(url=VIBRENT_FULFILLMENT_URL).valueString == 'shipped':
+        barcode_url = fhir_resource.extension.get(url=VIBRENT_BARCODE_URL).url
+      pid = fhir_resource.contained.get(
+        resourceType='Patient').identifier.get(system=VIBRENT_FHIR_URL + 'participantId')
+      p_id = from_client_participant_id(pid.value)
+    except AttributeError:
```

Comment:
Not sure we need this one, SimpleFhirR4Reader returns very explicit error messages that would be more helpful.

Reply:
I was testing with limited parameters and an un-handled attribute error was being thrown.  This is really to catch someone sending empty or nearly empty json.